src/crypto.rs

```diff
@@ -450,134 +570,13 @@ impl PrivateKey {
         }
     }
 
-    fn ed25519_from_pkcs8(der_key: &[u8]) -> Result<Self> {
-        Self::ed25519_from_pkcs8_with_keyid_hash_algorithms(
-            der_key,
-            python_tuf_compatibility_keyid_hash_algorithms(),
-        )
-    }
-
-    fn ed25519_from_pkcs8_with_keyid_hash_algorithms(
-        der_key: &[u8],
-        keyid_hash_algorithms: Option<Vec<String>>,
-    ) -> Result<Self> {
-        let key = Ed25519KeyPair::from_pkcs8(der_key)
-            .map_err(|_| Error::Encoding("Could not parse key as PKCS#8v2".into()))?;
-
-        let public = PublicKey::new(
-            KeyType::Ed25519,
-            SignatureScheme::Ed25519,
-            keyid_hash_algorithms,
-            key.public_key().as_ref().to_vec(),
-        )?;
-        let private = PrivateKeyType::Ed25519(key);
-
-        Ok(PrivateKey { private, public })
-    }
-
-    fn rsa_from_pkcs8(der_key: &[u8], scheme: SignatureScheme) -> Result<Self> {
-        if let SignatureScheme::Ed25519 = scheme {
-            return Err(Error::IllegalArgument(
-                "RSA keys do not support the Ed25519 signing scheme".into(),
-            ));
-        }
-
-        let key = RsaKeyPair::from_pkcs8(der_key)
-            .map_err(|_| Error::Encoding("Could not parse key as PKCS#8v2".into()))?;
-
-        if key.public_modulus_len() < 256 {
-            return Err(Error::IllegalArgument(format!(
-                "RSA public modulus must be 2048 or greater. Found {}",
-                key.public_modulus_len() * 8
-            )));
+    /// Unwrap the specific private key contained in this object and return it as a boxed trait
+    /// object.
+    pub fn into_boxed(self) -> Box<dyn PrivateKey> {
```

Comment:
We can leave the boxing up to the callers by implementing `PrivateKey` on this type, with:

```
impl PrivateKey for AnyPrivateKey {
    fn sign(&self, bytes: &[u8]) -> ... {
         match self {
             AnyPrivateKey::Ed25519(key) => key.sign(bytes),
             AnyPrivateKey::RSA(key) => key.sign(bytes)
     }
     ...
}
```

Reply:
Done.